src/main/java/org/elasticsearch/indices/recovery/RecoveryTarget.java

```diff
@@ -416,6 +417,12 @@ public void messageReceived(RecoveryCleanFilesRequest request, TransportChannel
 
     class FileChunkTransportRequestHandler extends BaseTransportRequestHandler<RecoveryFileChunkRequest> {
 
+        // When we last consulted RateLimiter:
+        long lastPauseCheckBytes;
```

Comment:
Oh good catch!  I thought this was for a single recovery, but this handler is in fact shared if multiple recoveries are running.  I'll switch to AtomicLongs